addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -2614,6 +2614,41 @@ export class OdooEditor extends EventTarget {
      *
      */
     initElementForEdition(element = this.editable) {
+        // Detect if the editable base element contain orphan inline nodes.
+        // If so we transform the base element HTML to put those orphans
+        // inside `<p>` containers.
+        const orphanInlineChildNodes = [...element.childNodes].filter(
+            (n) => !isBlock(n) && (n.nodeType === Node.ELEMENT_NODE || n.textContent.trim() !== "")
+        );
+        if (orphanInlineChildNodes.length) {
+            let currentP;
+            const childNodes = [...element.childNodes];
+            const tempEl = document.createElement('temp-container');
+            const appendAndRenewCurrentP = () => {
+                if (currentP && currentP.childNodes.length > 0 && currentP.innerHTML.trim() !== '') {
+                    tempEl.append(currentP);
+                }
+                currentP = document.createElement('p');
+                currentP.style.marginBottom = '0';
+            };
+            appendAndRenewCurrentP();
+            for (const node of childNodes) {
+                if (node.nodeType === Node.ELEMENT_NODE && node.tagName === 'BR') {
+                    if (!currentP.childNodes.length) {
+                        currentP.append(node);
+                    }
+                    appendAndRenewCurrentP();
+                } else if (isBlock(node)) {
+                    appendAndRenewCurrentP();
+                    tempEl.append(node);
+                } else {
+                    currentP.append(node);
+                }
```

Comment:
I'm pretty sure it's possible to write this without resorting to a local function with closure variables like `appendAndRenewCurrentP`. Since we are going to discuss other things (like why do we need to append the BR into the P), we can think together about this too.